Indexing/IndexManager.hpp

```diff
@@ -47,7 +47,6 @@ enum IndexType {
 
   DEMODULATION_SUBTERM_SUBST_TREE,
   DEMODULATION_LHS_CODE_TREE,
-  DEMODULATION_LHS_SUBST_TREE,
```

Comment:
No more `DEMODULATION_LHS_SUBST_TREE`? I'm happy if so, just wondering why it died.

Reply:
Since for this particular case we assert `ResultSubstitution::isIdentityOnQueryWhenResultBound()`, so implicitly asserting the use of `DEMODULATION_LHS_CODE_TREE`, I figured it's better to remove it. Otherwise, something like this can exist with substitution trees as well if needed, just not explicitly for demodulation.